tests/build/ is never collected — pytest's default norecursedirs excludes it

Found while working #653's Q4 (PR #775). Unrelated to that change; pre-existing.

## The problem

`tests/build/` is **never collected by a bare `pytest`**. Pytest's default `norecursedirs` is:

```
*.egg .* _darcs build CVS dist node_modules venv {arch}
```

`build` is in that list, and CLM does not override `norecursedirs` in `pyproject.toml`. So the directory name shadows the test package.

Verified on the current tree:

```console
$ pytest --collect-only -q | grep -c "tests/build"
0
```

The pre-push hook and all four CI test jobs run a bare `pytest`, so **nothing in `tests/build/` has been running**.

## How it surfaced

PR #775 removed a private validator helper. The full suite stayed green through a full adversarial review — but `tests/build/test_split_sources.py` imports that helper, so:

```console
$ pytest tests/build/
E   ImportError: cannot import name '_check_shared_cell_parity' from 'clm.slides.validator'
!!!!!! Interrupted: 1 error during collection !!!!!!
```

A broken test module that no gate could see. (PR #775 was rescoped for other reasons and no longer removes that symbol, so the import is fine again — but the hole is not.)

## What is in there

`tests/build/test_split_sources.py` — includes the only tests asserting the shared-cell **count mismatch** and byte-divergence messages. Worth checking whether anything else lives there and whether it still passes once collected.

## Fix

Set `norecursedirs` explicitly in `pyproject.toml`'s pytest config so the default list does not shadow a real test directory:

```toml
[tool.pytest.ini_options]
norecursedirs = ["*.egg", ".*", "_darcs", "CVS", "dist", "node_modules", "venv", "{arch}"]
```

(i.e. the default minus `build`.) Then run `pytest tests/build/` and fix whatever has rotted while invisible — expect some, since these tests have not run for an unknown period.

Renaming the directory (`tests/build_pipeline/`) is the alternative and avoids future surprise, at the cost of a path change.

## Worth checking at the same time

Whether any other directory name in `tests/` collides with the default list — `dist`, `node_modules`, or anything starting with `.` or matching `*.egg`.

--- a/changelog.d/776-pytest-collects-tests-build.fixed.md
+++ b/changelog.d/776-pytest-collects-tests-build.fixed.md
@@ -1,0 +1,6 @@
+- **`tests/build/` is collected again.** Pytest's default `norecursedirs`
+  contains `build`, so the `tests/build/` test package (split-source build
+  routing, shared-cell parity messages) was silently never collected by a bare
+  `pytest` — neither the pre-push hook nor any CI job ran it. `pyproject.toml`
+  now pins `norecursedirs` to the pytest default minus `build`; the 14
+  previously-invisible tests pass unchanged. (#776)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -238,6 +238,11 @@ include = [
 asyncio_mode = "auto"
 asyncio_default_fixture_loop_scope = "function"
 testpaths = ["tests"]
+# Pytest's DEFAULT norecursedirs contains ``build``, which silently shadowed
+# the real test package ``tests/build/`` — nothing in it was ever collected by
+# a bare ``pytest`` (issue #776). This is the default list minus ``build``.
+# If you add a test directory, make sure its name is not on this list.
+norecursedirs = ["*.egg", ".*", "_darcs", "CVS", "dist", "node_modules", "venv", "{arch}"]
 python_files = ["test_*.py", "*_test.py"]
 python_classes = ["Test*"]
 python_functions = ["test_*"]

